File: cplusplus/src/SoundSystem.cpp
#include "dllimport.hpp"
#include <unicode_conversion.hpp>
#include "SoundSystem.hpp"

using namespace std;
using namespace uniconv;

void SoundSystem::PlaySound(u16string filepath, bool wait) {
    auto filepathPtr = U16StringToUnicode(filepath);

    // [TODO] -> implement
}

File: csharp/SoundSystem.cs
using System;
using NetCoreAudio;
using Cs;

namespace Cs 
{
    class SoundSystem {
        public static void PlaySound(string filepath, bool wait = false) {
            string fullpath = Path.GetFullPath(filepath);
            Player player = new Player();
            player.Play(filepath);
            TimeSpan shorttime = new TimeSpan(1);
            while (wait && player.Playing) Thread.Sleep(shorttime);
        }   
    }
}